### PR TITLE
doc api get list recommend product

### PR DESCRIPTION
**Pull request by at-trangle**
## Description
doc api get list recommend product
## Migrations
NO

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

**Comment by at-phuongvu** [comment]
LGTM

File: docs/api-references/product.md [hunk 1]
```diff
@@ -144,3 +144,78 @@ Get product comparision
     "code": 200
 }
 ```
+### `GET` Recommend Products
+```
+/api/products/{product}/recommend
```
Comment (at-quangtran) [inline docs/api-references/product.md]: /api/products/recommend?param=...
vdu : 
- /api/products/recommend?product_id=1
- /api/products/recommend?price=100000&category_id=1..
tuỳ vào mình đang ở trang nào để recommend cho phù hợp

File: docs/api-references/product.md [hunk 2]
```diff
@@ -144,3 +144,78 @@ Get product comparision
...
+##### Example
+| URL | Description |
+|---|---|
+| /api/products/1/recommend | Get list recommend Products with Id 1 |
+
+#### Response
+```json
+{
+    "result": [
+        {
```
Comment (at-quangtran) [inline docs/api-references/product.md]: recommend 2,3 sản phẩm tương tự

File: docs/api-references/product.md [hunk 3]
```diff
@@ -144,3 +144,118 @@ Get product comparision
...
+|---|---|---|
+| product | int | Get list recommend Products (require) |
+| category_id | int | id of category |
+| price | int | price recommend product |
+
+##### Example
+| URL | Description |
+|---|---|
+| /api/products/recommend/1?category=1 | Get list recommend Products in category_id 1 with Id 1 |
+| /api/products/recommend/1?price=1000 | Get list recommend Products in price >= 1000 with Id 1 |
```
Comment (at-phuongvu) [inline docs/api-references/product.md]: xem lại cái `route` này đi chứ t thấy nó ko đúng. `/api/products/recommend?param=value` chứ `recommend/1` thì t ko hiểu lắm

**Comment by at-phuongvu** [comment]
LGTM

File: docs/api-references/product.md [hunk 4]
```diff
@@ -144,3 +144,118 @@ Get product comparision
...
+|---|---|---|
+| product_id | int | Get list recommend Products (require) |
+| category_id | int | id of category |
+| price | int | price recommend product |
+
+##### Example
+| URL | Description |
+|---|---|
+| /api/products/recommend?product_id=1 | Get list recommend Products with product id 1 |
+| /api/products/recommend?price=1000&category=1 | Get list recommend Products in price >= 1000 with Id 1 |
```
Comment (at-quangtran) [inline docs/api-references/product.md]: category_id

**Pull request merged**